lib/chewy/query.rb

```diff
@@ -1004,7 +1014,7 @@ def _request
       @_request ||= begin
         request = criteria.request_body
         request.merge!(index: _indexes.map(&:index_name), type: _types.map(&:type_name))
-        request.merge!(search_type: options[:search_type]) if options[:search_type]
+        request.merge!(search_type: criteria.search_options[:search_type]) if criteria.search_options[:search_type]
```

Comment:
move this logic to `request_body` method, like `{body: ...}.merge(serch_options)`